opsdroid/connector/matrix/connector.py

```diff
@@ -414,45 +448,50 @@ async def _send_room_creation(self, creation_event):
     @register_event(events.RoomName)
     @ensure_room_id_and_send
     async def _send_room_name_set(self, name_event):
-        return await self.connection.set_room_name(name_event.target, name_event.name)
+        return await self.connection.room_put_state(
+            name_event.target, "m.room.name", name_event.name
+        )
 
     @register_event(events.RoomAddress)
     @ensure_room_id_and_send
     async def _send_room_address(self, address_event):
-        try:
-            return await self.connection.set_room_alias(
-                address_event.target, address_event.address
-            )
-        except MatrixRequestError as err:
-            if err.code == 409:
+        res = await self.connection.room_put_state(
+            address_event.target, "m.room.aliases", address_event.address
+        )
+
+        if isinstance(res, nio.RoomPutStateError):
+            print("Error setting room address: " + res.message)
+            if res.staus_code == 409:
                 _LOGGER.warning(
                     f"A room with the alias {address_event.address} already exists."
                 )
+        return res
 
     @register_event(events.JoinRoom)
     @ensure_room_id_and_send
     async def _send_join_room(self, join_event):
-        return await self.connection.join_room(join_event.target)
+        return await self.connection.join(join_event.target)
 
     @register_event(events.UserInvite)
     @ensure_room_id_and_send
     async def _send_user_invitation(self, invite_event):
-        try:
-            return await self.connection.invite_user(
-                invite_event.target, invite_event.user_id
-            )
-        except MatrixRequestError as err:
-            content = json.loads(err.content)
-            if err.code == 403 and "is already in the room" in content["error"]:
+        res = await self.connection.room_invite(
+            invite_event.target, invite_event.user_id
+        )
+
+        if isinstance(res, nio.RoomInviteError):
+            if res.status_code == 403 and "is already in the room" in res.message:
                 _LOGGER.info(
                     f"{invite_event.user_id} is already in the room, ignoring."
                 )
 
```

Comment:
As above we should throw the exception if we don't understand it.
```suggestion
        if isinstance(res, nio.RoomInviteError):
            if res.status_code == 403 and "is already in the room" in res.message:
                _LOGGER.info(
                    f"{invite_event.user_id} is already in the room, ignoring."
                )
            else:
                raise res
```